### PR TITLE
(MERGE WHEN READY TO PUBLISH SITE) Publish the site to iceberg.apache.org

### PR DESCRIPTION
**Pull request by samredai**
This updates the .asf.yaml file to add the publish directive to https://iceberg.apache.org.

### CODE REVIEW

File: .asf.yaml [hunk 1]
```diff
@@ -33,4 +33,5 @@ notifications:
     issues:       issues@iceberg.apache.org
     pullrequests: issues@iceberg.apache.org
 
-
+publish:
+    whoami:  asf-site
```
Comment (kbendick) [inline .asf.yaml:37]: Nit: a newline so we don’t get the red mark? I’m not sure how much people care but I don’t think see them in the main repo.

**Review by kbendick — APPROVED** [review]
This LGTM. Thanks @samredai!

**Review by kbendick — APPROVED** [review]

**Review by kbendick — APPROVED** [review]
This LGTM. Thanks @samredai!

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @samredai! Great work.

**Pull request merged**